src/components/Tabs/Scroller.tsx

```diff
@@ -0,0 +1,236 @@
+
+const styles = {
+  container: css({
+    position: 'relative',
+    overflow: 'hidden'
+  }),
+  breakoutMargin: css({
+    [mUp]: {
+      margin: 0
+    }
+  }),
+  scroller: css({
+    display: 'flex',
+    flexDirection: 'row',
+    overflowX: 'scroll',
+    flexWrap: 'nowrap',
+    scrollbarWidth: 'none' /* Firefox */,
+    msOverflowStyle: 'none' /* IE 10+ */,
+    WebkitOverflowScrolling: 'touch',
+    '::-webkit-scrollbar': {
+      display: 'none'
+    }
+  }),
+  arrow: css(plainButtonRule, {
+    display: 'none',
+    '@media (hover)': {
+      display: 'flex',
+      position: 'absolute',
+      top: 0,
+      bottom: 0,
+      textAlign: 'center',
+      alignItems: 'center',
+      justifyContent: 'center',
+      pointerEvents: 'none',
+      opacity: 0,
+      transition: 'opacity 200ms'
+    }
+  }),
+  arrowIcon: css({
+    // ontop of arrowBg
+    position: 'relative'
+  }),
+  arrowBg: css({
+    display: 'block',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0,
+    opacity: 0.7
+  }),
+  arrowHoverable: css({
+    '@media (hover)': {
+      '[role=group]:hover > &': {
+        pointerEvents: 'auto',
+        opacity: 1
+      }
+    }
+  })
+}
+
+type ScrollerType = {
+  children: React.ReactNode
+  center?: boolean
+  activeChildIndex?: number
+  hideArrows?: boolean
+  arrowSize?: number
+  breakoutPadding?: number
+}
+
+const Scroller = ({
+  children,
+  activeChildIndex = 0,
+  center = false,
+  hideArrows = false,
+  arrowSize = 28,
+  breakoutPadding = 0
+}: ScrollerType) => {
+  const scrollRef = useRef<HTMLDivElement>()
+  const [{ left, right }, setArrows] = useState({ left: false, right: false })
+  const [scrollerOverflow, setOverflow] = useState(false)
+  const [colorScheme] = useColorContext()
+
+  useEffect(() => {
+    const scroller = scrollRef.current
+    if (!scroller) return
+    const target = Array.from(scroller?.children)[activeChildIndex + 1] // + 1 for pad element
+    scrollIntoView(target, {
+      time: 400,
+      align: {
+        left: 0,
+        leftOffset: breakoutPadding,
+        ...getTop()
+      }
+    })
+  }, [activeChildIndex])
+
+  useEffect(() => {
+    const scroller = scrollRef.current
+
+    const measure = () => {
+      let left = false
+      let right = false
+      if (scroller.scrollLeft > 0) {
+        left = true
+      }
+      if (scroller.scrollLeft + scroller.clientWidth < scroller.scrollWidth) {
+        right = true
+      }
+      setArrows(current => {
+        if (current.left !== left || current.right !== right) {
+          return { left, right }
+        }
+        return current
+      })
+      setOverflow(scroller.clientWidth < scroller.scrollWidth)
+    }
+    scroller.addEventListener('scroll', measure)
+    window.addEventListener('resize', measure)
+    measure()
+    return () => {
+      scroller.removeEventListener('scroll', measure)
+      window.removeEventListener('resize', measure)
+    }
+  }, [])
```

Comment:
Ah I overlooked that the scroll event is listening to the `scroller` itself. I guess moving the assignment outside of the effect and adding `scroller` to the dependency would be the 100% correct solution to ensure that the listener is always attached to the current dom element. But probably the dom element never changes in practice right now. 

```suggestion
  const scroller = scrollRef.current
  useEffect(() => {
    const measure = () => {
      let left = false
      let right = false
      if (scroller.scrollLeft > 0) {
        left = true
      }
      if (scroller.scrollLeft + scroller.clientWidth < scroller.scrollWidth) {
        right = true
      }
      setArrows(current => {
        if (current.left !== left || current.right !== right) {
          return { left, right }
        }
        return current
      })
      setOverflow(scroller.clientWidth < scroller.scrollWidth)
    }
    scroller.addEventListener('scroll', measure)
    window.addEventListener('resize', measure)
    measure()
    return () => {
      scroller.removeEventListener('scroll', measure)
      window.removeEventListener('resize', measure)
    }
  }, [scroller])
```